wallet-new/src/Cardano/Wallet/Kernel/BListener.hs

```diff
@@ -250,7 +250,7 @@ applyBlock pw@PassiveWallet{..} b = do
         else do
             rb   <- hashToBlock (cur ^. bcHash . fromDb)
             prev <- traverse hashToBlock (rb ^? rbContext . bcPrevMain . _Just . fromDb)
-            findMissing (prev ^? _Just . rbContext) tgt (rb : acc)
+            rb `seq` findMissing (prev ^? _Just . rbContext) tgt (rb : acc)
```

Comment:
Are we sure we do want to force `rb` to whnf? I suspect a more useful forcing would be one on the `acc` (basically by having a `!` in front of `acc` so that we avoid accumulating lazy thunks but we keep the _spine_ of the list strict. what am I missing? Essentially this fix seems to force the values of the data structure to be in whnf but the spine is still lazy, which can be problematic if `acc` turns out to be large enough.